.github/code/nipoppy_track_processing_for_dataset.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+set -euo pipefail
+
+DS_ID=$1
+DS_URL="https://github.com/OpenNeuroDatasets/${DS_ID}.git"
+
+SCRIPT_DIR="$(dirname "$(realpath "$0")")"
+DERIVATIVES_REPO_MAP="derivatives/repo_derivatives_map.json"
+SOURCE_WORKDIR="derivatives/source"
+NIPOPPY_WORKDIR="derivatives/nipoppy"
+TEMPLATE_CONFIGS_DIR="derivatives/template_pipeline_configs"
+NIPOPPY_DATASET_DIR="${NIPOPPY_WORKDIR}/${DS_ID}"
+OUTPUT_DIR="processing_status_files"
+
+mkdir -p ${SOURCE_WORKDIR}
+mkdir -p ${NIPOPPY_WORKDIR}
+
+git clone -q ${DS_URL} ${SOURCE_WORKDIR}/${DS_ID}
+echo "${DS_ID}: Initializing Nipoppy dataset"
+# NOTE: The BIDS dataset is symlinked by default
+nipoppy init --dataset ${NIPOPPY_DATASET_DIR} --bids-source ${SOURCE_WORKDIR}/${DS_ID}
+
+# This creates a multi-line string variable
+derivative_datasets="$(jq -r --arg ds "$DS_ID" '.[ $ds ] | keys[]' "$DERIVATIVES_REPO_MAP")"
+
+has_no_ses=0
+if ! python ${SCRIPT_DIR}/does_dataset_have_sessions.py "$DS_ID" "$NIPOPPY_DATASET_DIR/bids"; then
+    has_no_ses=1
+    echo "${DS_ID}: No sessions found."
+fi
+
+while IFS= read -r derivative_ds; do
+    pipeline_info="$(jq -c --arg ds "$DS_ID" --arg repo "$derivative_ds" '.[ $ds ][ $repo ]' "$DERIVATIVES_REPO_MAP")"
+
+    # Check that the derivative dataset has pipeline info
+    if [ "$pipeline_info" != "{}" ]; then
+        echo "${DS_ID}: ${derivative_ds}: Cloning derivative dataset"
+        derivative_ds_url="https://github.com/OpenNeuroDerivatives/${derivative_ds}.git"
+        pipeline_name=$(jq -r '.name | ascii_downcase' <<< "$pipeline_info")
+        pipeline_version=$(jq -r '.version' <<< "$pipeline_info")
+        pipeline_dir="${NIPOPPY_DATASET_DIR}/derivatives/${pipeline_name}/${pipeline_version}/output"
+        mkdir -p "${pipeline_dir}"
+        git clone -q "${derivative_ds_url}" "${pipeline_dir}"
+
+        pipeline_config_dir="${NIPOPPY_DATASET_DIR}/pipelines/processing/${pipeline_name}-${pipeline_version}"
+        mkdir -p "${pipeline_config_dir}"
+
+        args=(
+            "${TEMPLATE_CONFIGS_DIR}/${pipeline_name}/config.json"
+            "${TEMPLATE_CONFIGS_DIR}/${pipeline_name}/tracker.json"
+            "${pipeline_version}"
+            "${pipeline_config_dir}"
+        )
+        if [ $has_no_ses -eq 1 ]; then
+            args+=("--no-sessions")
+        fi
+        echo "${DS_ID}: ${derivative_ds}: Customizing config.json and tracker.json files"
+        python ${SCRIPT_DIR}/customize_configs_for_pipeline.py "${args[@]}"
```

Comment:
Mentioned this above, but that's a lot of logic to handle session `yes/no`. Is there another good reason to have a config customizer right now? Otherwise I'd prefer we just have templates with and without sessions for each pipeline - and then address the session issue long-term on the Nipoppy side